src/RequestProcessor.cs

```diff
@@ -281,6 +337,151 @@ private void BindOutputFromResult(InvocationResponse response, AzFunctionInfo fu
             }
         }
 
+        /// <summary>
+        /// Installs managed dependencies modules specified in Requirements.psd1 for the function app.
+        /// If an exception is raised during installation, wrap it in a DependencyInstallationException.
+        /// This exception will get unwrapped in caller method.
+        /// </summary>
+        private void InstallManagedDependencyModule(System.Management.Automation.PowerShell pwsh, ILogger logger)
+        {
+            try
+            {
+                RunInstallManagedDependencyModule(pwsh, logger);
+                DependencyManager.FunctionAppDependenciesInstalled = true;
+            }
+            catch (Exception e)
+            {
+                var errorMsg = string.Format(PowerShellWorkerStrings.FailToInstallFuncAppDependencies);
+                var dependencyInstallationException = new DependencyInstallationException(errorMsg, e);
+                throw dependencyInstallationException;
+            }
+        }
+
+        /// <summary>
+        /// Installs managed dependencies modules specified in Requirements.psd1 for the function app.
+        /// </summary>
+        private void RunInstallManagedDependencyModule(System.Management.Automation.PowerShell pwsh, ILogger logger)
```

Comment:
This needs to be made testable by not having this as private method, as there're lots of logic inside this function

Reply:
@SatishRanjan : Moving this code to the DependencyManager. Also, this code will be tested by a couple of end to end scenarios I will add.